src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,44 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private int size;
+    private Node<T> current;
 
+    public void push(T value) {
+        current = new Node<>(current, value);
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (current == null) {
+            throw new EmptyStackException();
+        }
```

Comment:
Good job, actually I think you can create separate method for this check, because you used it several times)